src/components/GoodsList/GoodsList.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { GoodsItem } from '../GoodsItem/GoodsItem';
+
+import './GoodsList.css';
+
+export class GoodsList extends React.Component {
+  state = {
+    hideList: true,
+    goodsList: [...this.props.goodsList],
+    goodsLength: 1,
+  }
+
+  showList = () => {
+    this.setState({
+      hideList: false,
+    });
+  }
+
+  reverseList = () => {
+    this.setState(({ goodsList }) => ({
+      goodsList: [...goodsList].reverse(),
+    }));
+  }
+
+  sortAlphabet = () => {
+    this.setState(({ goodsList }) => ({
+      goodsList: [...goodsList].sort((a, b) => a.localeCompare(b)),
+    }));
+  }
+
+  resetList = () => {
+    this.setState({
+      goodsList: [...this.props.goodsList],
+      goodsLength: 1,
+    });
+  }
+
+  sortByLength = () => {
+    this.setState(({ goodsList }) => ({
+      goodsList: [...goodsList].sort((a, b) => a.length - b.length),
+    }));
+  }
+
+  selectByLength = (length) => {
+    this.setState({
+      goodsLength: length,
+      goodsList: [...this.props.goodsList]
+        .filter(good => good.length >= length),
```

Comment:
filter не мутирует исходный массив, не надо делать копию
```suggestion
      goodsList: this.props.goodsList
        .filter(good => good.length >= length),
```

Reply:
исправил.